### DISCUSSION TITLE
Is there a way to determine (deep) memory usage of Dataset or DataArray objects?

### DISCUSSION THREAD

**Discussion by johann-petrak** [pydata/xarray#7536] [Q&A] [answered]
The only thing I  found is CFTimeIndex.memory_usage but nothing equivalent for Dataset or DataArray. Is there some other method?

**Comment by nohcnamp** [comment]
Hello,

You can use the `nbytes` method. This method will return the size in bytes of your object. It works for `Dataset` and `DataArray` objetcs. I'm not sure for `DatasetGroupBy` and `DataArrayGroupBy` objects though.
From there you'll be able to convert it to Mb, Gb, etc. There is [this nice stackoverflow thread](https://stackoverflow.com/questions/5194057/better-way-to-convert-file-sizes-in-python) covering this subject.

Pierre

**Reply by johann-petrak** [reply]
Thanks, but it seems this just passes through the value for the underlying numpy array of a DataArray, it does not include whatever the overhead of the DataArray structure is, nor the size needed by the coordinates, attributes etc.

**Reply by nohcnamp** [reply]
> Thanks, but it seems this just passes through the value for the underlying numpy array of a DataArray, it does not include whatever the overhead of the DataArray structure is, nor the size needed by the coordinates, attributes etc.

I misunderstood what you meant, sorry.

**Comment by dcherian** [comment, marked as answer]
See #5764 . There's no way to do it at present.
